gap/attr.gi

```diff
@@ -954,6 +954,34 @@ function(digraph)
   return girth;
 end);
 
+InstallMethod(DigraphOddGirth, "for a digraph",
+[IsDigraph],
+function(digraph)
+  local A, B, k, N, girth, powerfound;
+  girth := DigraphGirth(digraph);
+  if girth = infinity or IsOddInt(girth) then
+    return girth;
+  fi;
+  N          := DigraphNrVertices(digraph);
+  A          := AdjacencyMatrix(digraph);
+  B          := Immutable(A ^ 2);
+  k          := 1;
```

Comment:
If the girth is (say) `6`, then the odd girth is at least `7`, right? So why don't we skip checking whether the girth is `1`, `3`, or `5`?

Reply:
Good comment!